src/modules/dashboard/components/Extensions/utils.ts

```diff
@@ -28,3 +30,19 @@ export const createExtensionDefaultValues = (
     return defaultValues;
   }, {});
 };
+
+export const getButtonAction = (
+  actionType: 'SUBMIT' | 'ERROR' | 'SUCCESS',
+  extensionId: string,
+) => {
+  const actionEnd = actionType === 'SUBMIT' ? '' : `_${actionType}`;
+  let actionBeginning = 'COLONY_EXTENSION';
+
+  if (extensionId === Extension.CoinMachine) {
+    actionBeginning = 'COIN_MACHINE';
+  } else if (extensionId === Extension.Whitelist) {
+    actionBeginning = 'WHITELIST';
+  }
```

Comment:
Maybe this could be a switch, to account for more extensions in future.

Or, better yet, do we have to manually assign a string here? Can we use the string from `Extension.CoinMachine` and use that in ActionTypes as well? I'm sure there is a reason for this, just checking though.

Reply:
Yeah, this could be a good place for a switch

About using `Extension.CoinMachine` for the ActionType, It's not possible to use directly because the format of the ActionTypes is like `COIN_MACHINE_START_THE_PARTY`, so we use snake case + uppercase, while the format that comes with `Extension.` is camel case with capitalization (`CoinMachine`). We would need to use 1 or 2 functions first to format them like we format our ActionTypes, which may not be worth it